plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -53,34 +59,89 @@ public void configure(Binder binder)
         configBinder(binder).bindConfig(OracleConfig.class);
         newOptionalBinder(binder, Key.get(int.class, MaxDomainCompactionThreshold.class)).setBinding().toInstance(ORACLE_MAX_LIST_EXPRESSIONS);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, RetryStrategy.class).setBinding().to(OracleRetryStrategy.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, Key.get(ConnectionFactory.class, ForRetryingConnectionFactory.class))
+                .setBinding()
+                .toProvider(OracleConnectionFactoryProvider.class)
+                .in(Scopes.SINGLETON);
     }
 
-    @Provides
-    @Singleton
-    @ForBaseJdbc
-    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig, OpenTelemetry openTelemetry)
-            throws SQLException
+    private static class OracleRetryStrategy
+            implements RetryStrategy
     {
-        Properties connectionProperties = new Properties();
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_INCLUDE_SYNONYMS, String.valueOf(oracleConfig.isSynonymsEnabled()));
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_REPORT_REMARKS, String.valueOf(oracleConfig.isRemarksReportingEnabled()));
+        @Override
+        public boolean isExceptionRecoverable(Throwable exception)
+        {
+            return Throwables.getCausalChain(exception).stream()
+                    .anyMatch(SQLRecoverableException.class::isInstance);
+        }
+    }
 
-        if (oracleConfig.isConnectionPoolEnabled()) {
-            return new OraclePoolConnectionFactory(
+    private static class OracleConnectionFactoryProvider
```

Comment:
why not to use factory method like below?

Reply:
That was the 'previous' approach before we changed the acceptance criteria. This code is now deleted.